packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx

```diff
@@ -0,0 +1,43 @@
+import { css, keyframes } from '@emotion/css';
+import React from 'react';
+
+/**
+ * @internal
+ */
+export interface LoadingBarProps {
+  containerWidth: number;
+  width?: number;
+  height?: number;
+  ariaLabel?: string;
+}
+
+/**
+ * @internal
+ */
+export const LoadingBar: React.FC<LoadingBarProps> = ({ containerWidth, width, height, ariaLabel = 'Loading bar' }) => {
+  const loadingStyles = getLoadingStyles(containerWidth, width, height);
+
+  return <div aria-label={ariaLabel} className={loadingStyles.loading}></div>;
+};
+
+const getLoadingStyles = (containerWidth: number, width?: number, height?: number) => {
+  const loadingWidth = width ?? 128;
+  const loadingAnimation = keyframes({
+    '0%': {
+      transform: 'translateX(0)',
+    },
+    '100%': {
+      transform: `translateX(${containerWidth - loadingWidth}px)`,
+    },
+  });
+  return {
+    loading: css({
+      width: `${loadingWidth}px`,
```

Comment:
This doesn't allow specifying width in percentages. For example, in the story, if I set the container's width to 100%, the loading bar doesn't move all the way to the right. Maybe this component needs an extra wrapper with width 100% and then the parent components can modify the width if needed?